### PR TITLE
[DRK Sim] Improve Living Shadow, add Prepull TBN, misc improvements/fixes

### PR DESCRIPTION
**Pull request by VioletHynes**
Living Shadow now works correctly:
- Damage is applied when it should be applied
- Darkside no longer applies
- It still needs to get the correct stats, but that will come later, as it requires a deeper change

Prepull TBN is added

Some extra pre-work has been done in advance of adding the computationally created rotation

image

### CODE REVIEW

File: packages/core/src/sims/tank/drk/drk_actions.ts [hunk 1]
```diff
@@ -210,4 +197,91 @@ export const Shadowbringer: DrkOgcdAbility = {
...
+        charges: 1,
+    },
+};
+
+export const LivingShadowShadowstride: DrkOgcdAbility = {
+    type: 'ogcd',
+    name: "(Living Shadow) Shadowstride",
+    animationLock: 0,
+    id: 38512,
+    potency: 0,
```
[the remark this answers is not in the document]
Comment (VioletHynes) [reply packages/core/src/sims/tank/drk/drk_actions.ts:240]: `null` vs 0 potency is subtly different -- 0 potency will start the fight if it's not started (e.g. Shadowstride), whereas `null` potency won't (e.g. TBN)

File: packages/core/src/sims/tank/drk/drk_lv100_sim.ts [hunk 2]
```diff
@@ -52,18 +56,27 @@ Results are only currently accurate up to 9 minutes (540s).`,
     }],
 };
 
+// LivingShadowAbilityUsageTime is a type representing two
+// things: a Living Shadow ability and the time it should go off at
+type LivingShadowAbilityUsageTime = {
+    // The Living Shadow ability to use
+    ability: DrkAbility,
```
[the remark this answers is not in the document]
Comment (VioletHynes) [reply packages/core/src/sims/tank/drk/drk_lv100_sim.ts:63]: Yeah, that's a good thought

File: packages/core/src/sims/tank/drk/drk_lv100_sim.ts [hunk 3]
```diff
@@ -197,7 +337,40 @@ export class DrkSim extends BaseMultiCycleSim<DrkSimResult, DrkSettings, DrkCycl
         return abilityUseResult;
     }
 
-    static getRotationForGcd(gcd: number): DrkRotationData {
+    private useOpener(cp: DrkCycleProcessor, prepullTBN: boolean) {
+        if (prepullTBN) {
+            this.use(cp, Actions.TheBlackestNight)
+            cp.advanceTo(3 - STANDARD_ANIMATION_LOCK);
+            // Hacky out of combat mana tick.
+            cp.gauge.magicPoints += 600
```
[the remark this answers is not in the document]
Comment (VioletHynes) [reply packages/core/src/sims/tank/drk/drk_lv100_sim.ts:345]: Good call

File: packages/core/src/sims/tank/drk/drk_lv100_sim.ts [hunk 4]
```diff
@@ -72,11 +85,62 @@ class DrkCycleProcessor extends CycleProcessor {
...
+        let darksideDuration = 0
+        // Get what the the Darkside duration would be at this
+        // point of time, for visualization in the UI
+        const darkside = buffs.find(buff => buff.name === "Darkside")
+        if (darkside) {
+            const buffData = this.getActiveBuffData(darkside, abilityUsage.usageTime)
+            if (buffData) {
+                darksideDuration = Math.round(buffData.end - abilityUsage.usageTime)
+            }
+        }
```
[the remark this answers is not in the document]
Comment (VioletHynes) [reply packages/core/src/sims/tank/drk/drk_lv100_sim.ts:114]: Yeah, this is possible to one-line, but I felt it was cleaner this way. I think I like removing the `if darkside` for the `&&`, so I'll do that

**Review by amakiirs — APPROVED** [review]

**Pull request merged**